spec/services/mixpanel_service_spec.rb

```diff
@@ -254,154 +254,158 @@
+                    client_site_name: "Child Site",
+                    client_site_id: client.vita_partner.id,
+                    user_id: user.id,
+                    user_site_name: site.name,
+                    user_site_id: site.id,
+                    user_organization_name: organization.name,
+                    user_organization_id: organization.id,
+                    user_coalition_name: coalition.name,
+                    user_coalition_id: coalition.id,
+                    days_since_ready_for_prep: 1,
+                    hours_since_ready_for_prep: 28,
+                    days_since_tax_return_created: 2,
+                    hours_since_tax_return_created: 48
+                }
+            )
+          end
         end
-      end
-
-      context "when the event is triggered by the system" do
-        let(:tax_return) { create :tax_return, certification_level: "basic", client: client }
 
-        before do
-          tax_return.update_column(:ready_for_prep_at, 28.hours.ago)
-          tax_return.update_column(:created_at, 2.days.ago)
-        end
+        context "when the event is triggered by the system" do
+          let(:tax_return) { create :tax_return, certification_level: "basic", client: client }
 
-        it "handles the lack of a last_changed_by user" do
-          MixpanelService.send_file_rejected_event(tax_return)
+          before do
+            tax_return.transition_to(:prep_ready_for_prep)
+            TaxReturnTransition.where(to_state: "prep_ready_for_prep", tax_return_id: tax_return.id).update(created_at: 28.hours.ago)
+            tax_return.update_column(:created_at, 2.days.ago)
+          end
 
-          expect(fake_tracker).to have_received(:track).with(
-            "fake_visitor_id",
-            "filing_rejected",
-            hash_excluding(
-              {
-                user_id: user.id,
-                user_site_name: site.name,
-                user_site_id: site.id,
-                user_organization_name: organization.name,
-                user_organization_id: organization.id,
-                user_coalition_name: coalition.name,
-                user_coalition_id: coalition.id,
-              }
+          it "handles the lack of a last_changed_by user" do
+            MixpanelService.send_file_completed_event(tax_return, "filing_rejected")
+
+            expect(fake_tracker).to have_received(:track).with(
+                "fake_visitor_id",
+                "filing_rejected",
+                hash_excluding(
+                    {
+                        user_id: user.id,
+                        user_site_name: site.name,
+                        user_site_id: site.id,
+                        user_organization_name: organization.name,
+                        user_organization_id: organization.id,
+                        user_coalition_name: coalition.name,
+                        user_coalition_id: coalition.id,
```

Comment:
if you feel so inclined, it might be better to make this a positive rather than a negative test (so, like the above, would just have the hash that _is_ sent). it is a bit longer / redundant though and it was already like this so totally fine if you don't feel like changing it!